source/MapPanel.cpp

```diff
@@ -1173,28 +1192,38 @@ void MapPanel::DrawTooltips()
 	// Create the tooltip text.
 	if(tooltip.empty())
 	{
-		pair<pair<int, int>, int> t = escortSystems.at(hoverSystem);
+		MapPanel::SystemTooltipData t = escortSystems.at(hoverSystem);
+
 		if(hoverSystem == &playerSystem)
 		{
-			--t.first.first;
-			if(t.first.first || t.first.second || t.second)
+			--t.activeShips;
```

Comment:
Not a bug you introduced, but I just realized that this can make activeShips negative if the player has no flagship. (And I just tested this, it shows -1 escorts if I open the map after I just sold my flagship with the original implementation.)